src/Common/ZooKeeper/ZooKeeperImpl.cpp

```diff
@@ -358,6 +358,8 @@ ZooKeeper::ZooKeeper(
         initFeatureFlags();
         keeper_feature_flags.logFlags(log);
 
+        initAvailabilityZone();
```

Comment:
Consider to invoke this only local az load balancing option is used.

Reply:
Decide not to do. Similar to api flag. separation of concern as well. existing integration test would go through this flow to ensure no issue.